storage_service/locations/models/dspace_rest.py

```
@@ -0,0 +1,398 @@
+                            self.as_url.port,
+                            self.as_user,
+                            self.as_repository)
+
+        try:
+            client.add_digital_object('/repositories/' + self.as_repository + '/archival_objects/' +
+                                      archivesspace_collection,
+                                      package_uuid,
+                                      title=package_title,
+                                      uri=self.ds_rest_url.scheme + '://' + self.ds_rest_url.netloc + '/handle/' + dspace_item['handle'])
+        except RequestException as e:
+            LOGGER.error('Could not add digital object to ArchivesSpace')
+            if e.response and e.response.status_code == 404:
+                raise ValueError('Either repository %s or archival object %s does not exist',
+                                 14, archivesspace_collection)
+
+            raise ValueError('ArchivesSpace Server error: %s', e.response.text)
+
+        LOGGER.info('ArchivesSpace Client: %s', client)
+
+    def _deposit_aip_to_dspace(self, source_path, headers, dspace_item, dspace_sessionid):
+        LOGGER.info('Package is AIP. Sending file.')
+
+        bitstream_url = self.ds_rest_url.scheme + '://' + self.ds_rest_url.netloc + self.ds_rest_url.path + '/items/'
+        bitstream_url += dspace_item['uuid'] + '/bitstreams?name=' + urllib.quote(
+            os.path.basename(source_path).encode('utf-8'))
+
+        try:
+            with open(source_path, 'rb') as content:
+                requests.post(bitstream_url,
+                              headers=headers,
+                              cookies={'JSESSIONID': dspace_sessionid},
+                              data=content,
+                              verify=self.verify_ssl)
+            LOGGER.debug('%s successfully sent to %s', source_path, bitstream_url)
+        except Exception:
+            LOGGER.info('Error sending %s, to %s', source_path, bitstream_url)
```

This seems like a significant failure if we get to this point. I think you should be using `LOGGER.error` here. More importantly, I don't think the exception should be swallowed here. We need to make sure it is raised so that the request to store the AIP (or DIP) fails when one of the crucial steps fails. I think you need to go through the code and a) make sure that the log statements are the appropriate level and b) make sure that fatal exceptions are not being suppressed but are being allowed to bubble up so that Archivematica will know when the request to store the AIP has failed. If it is important to make sure that we make the call to logout from DSpace no matter what, then you could put that code in a finally block and any fatal exception will still get raised, for example:

    >>> try:
    ...   print(dog)
    ... except NameError:
    ...   print('no dog')
    ...   raise
    ... finally:
    ...   print('done')
    ... 
    no dog
    done
    Traceback (most recent call last):
      File "<stdin>", line 2, in <module>
    NameError: name 'dog' is not defined

---

Yeah. I'll be the first to admit that exception handling is sloppy.

I'll do just what you suggest.